Event/AuthenticationFailureEvent.php

```diff
@@ -24,10 +24,16 @@ class AuthenticationFailureEvent extends Event
      */
     protected $response;
 
-    public function __construct(AuthenticationException $exception, Response $response)
+    /**
+     * @var Request
+     */
+    protected $request;
+
+    public function __construct(AuthenticationException $exception, Response $response, Request $request)
```

Comment:
```suggestion
    public function __construct(AuthenticationException $exception, Response $response, ?Request $request = null)
```

BC break otherwise :)

Reply:
updated